internal/services/task_status_listener.go

```diff
@@ -121,8 +123,16 @@ func (i *TaskStatusListener) processSmartcarPollStatusEvent(event *shared.CloudE
 	}
 
 	udai.Status = models.UserDeviceAPIIntegrationStatusAuthenticationFailure
-	_, err = udai.Update(context.Background(), i.db().Writer, boil.Infer())
-	return err
+	if _, err = udai.Update(context.Background(), i.db().Writer, boil.Infer()); err != nil {
+		i.log.Err(err).Str("userDeviceID", userDeviceID).Str("integrationID", integrationID).Msg("failed up update user device api integration with failure status")
+	}
+
+	vehicleID, ok := udai.R.UserDevice.TokenID.Uint64()
+	if !ok {
+		i.log.Err(errors.New("failed to parse vehicle token id for cio event"))
```

Comment:
Do we want to return here as well or do we want to send a Disconnection event with out a vehilceId

Reply:
updated to return